File: 1-1000/0824-goat-latin/0824-goat-latin.cpp
class Solution {
public:
    string toGoatLatin(string sentence) {
        string vowels = "aeiouAEIOU";
        stringstream ss(sentence);
        string word, res;
        int idx = 1;
        while (ss >> word) {
            if (vowels.find(word[0]) != string::npos) {
                res += word;
            } else {
                res += word.substr(1) + word[0];
            }
            res += "ma";
            res += string(idx, 'a');
            res += " ";
            idx++;
        }
        res.pop_back();
        return res;
    }
};

File: 1-1000/0824-goat-latin/0824-goat-latin.java
class Solution {
    public String toGoatLatin(String sentence) {
        String[] words = sentence.split(" ");
        String vowels = "aeiouAEIOU";
        StringBuilder sb = new StringBuilder();
        int idx = 1;

        for (String w : words) {
            if (vowels.indexOf(w.charAt(0)) != -1) {
                sb.append(w);
            } else {
                sb.append(w.substring(1)).append(w.charAt(0));
            }

            sb.append("ma");
            sb.append("a".repeat(idx));
            sb.append(" ");
            idx++;
        }

        return sb.toString().trim();
    }
}